op-bindings/bindgen/etherscan-contracts.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+type etherscanContract struct {
+	Name             string
+	DeployedAddress  string
+	PredeployAddress string
+	Abi              string
+	Bytecode         string
+}
+
+type etherscanApiResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message"`
+	Result  string `json:"result"`
+}
+
+type etherscanContractMetadata struct {
+	Name              string
+	DeployedBin       string
+	Package           string
+}
+
+func readEtherscanContractsList(filePath string) ([]etherscanContract, error) {
+	var contracts []etherscanContract
+	err := readJSONFile(filePath, &contracts)
+	return contracts, err
+}
+
+func fetchEtherscanData(url string) (string, error) {
+	// TODO Handle rate limit error
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var response etherscanApiResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return "", err
+	}
+
+	return response.Result, nil
+}
+
+func writeEtherscanContractMetadata(contractMetaData etherscanContractMetadata, metadataOutputDir, contractName string, fileTemplate *template.Template) error {
+	metaDataFilePath := filepath.Join(metadataOutputDir, strings.ToLower(contractName)+"_more.go")
+	metadataFile, err := os.OpenFile(
+		metaDataFilePath,
+		os.O_RDWR|os.O_CREATE|os.O_TRUNC,
+		0o600,
+	)
+	if err != nil {
+		return fmt.Errorf("Error opening %s's metadata file at %s: %v", contractName, metaDataFilePath, err)
+	}
+
+	if err := fileTemplate.Execute(metadataFile, contractMetaData); err != nil {
+		return fmt.Errorf("Error writing %s's contract metadata at %s: %v", contractName, metaDataFilePath, err)
+	}
+	metadataFile.Close()
```

Comment:
You want to do `defer metadataFile.Close()` right after opening the file so its closed in all cases, it is never closed in the `err` case here